plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -237,8 +251,32 @@ func init() {
 	})
 }
 
-func createRealIngestor(client localClient, database string, tableName string) (localIngestor, error) {
-	ingestor, err := ingest.New(client.(*kusto.Client), database, tableName, ingest.WithStaticBuffer(bufferSize, maxBuffers))
+// For each table create the ingestor
+func createIngestorByTable(client *kusto.Client, database string, tableName string, ingestionType string) (ingest.Ingestor, error) {
+	var ingestor ingest.Ingestor
+	var err error
+	switch strings.ToLower(ingestionType) {
+	case managedIngestion:
+		{
+			mi, err := ingest.NewManaged(client, database, tableName)
+			if err != nil {
+				return nil, err
+			}
+			ingestor = mi
+		}
+	case queuedIngestion:
+		{
+			qi, err := ingest.New(client, database, tableName, ingest.WithStaticBuffer(bufferSize, maxBuffers))
+			if err != nil {
+				return nil, err
+			}
+			ingestor = qi
+		}
+	default:
+		{
+			err = fmt.Errorf(`ingestion_type has to be one of %q or %q`, managedIngestion, queuedIngestion)
```

Comment:
Please directly return the error here instead of risking that it get's overwritten later!